Glamourer/Api/GlamourerIpc.GetDesign.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Buffers.Text;
+using System.Collections.Generic;
+using System.Linq;
+using Dalamud.Game.ClientState.Objects.Types;
+using Dalamud.Plugin;
+using Glamourer.Customization;
+using Glamourer.Designs;
+using Glamourer.Structs;
+using Penumbra.Api.Helpers;
+using Penumbra.GameData.Actors;
+
+namespace Glamourer.Api;
+
+public partial class GlamourerIpc
+{
+    public const string LabelGetDesignList = "Glamourer.GetDesignList";
+
+    private readonly FuncProvider<DesignListEntry[]> _getDesignListProvider;
+
+    public static FuncSubscriber<DesignListEntry[]> GetDesignListSubscriber(DalamudPluginInterface pi)
+        => new(pi, LabelGetDesignList);
+
+    public DesignListEntry[] GetDesignList()
+        => _designManager.Designs.Select(x => new DesignListEntry(x)).ToArray();
+
+    public record class DesignListEntry
```

Comment:
It's a simple pair of Dalamud-known types, if you just return a pair in IPC it will not go through ser/deser over json for no reason, just `(string Name, Guid Identifier)`